plan/exhaust_physical_plans.go

```diff
@@ -541,33 +545,33 @@ func (la *LogicalAggregation) getStreamAggs(prop *requiredProp) []PhysicalPlan {
 	if len(la.groupByCols) != len(la.GroupByItems) {
 		return nil
 	}
-	streamAggs := make([]PhysicalPlan, 0, len(la.possibleProperties)*2)
-	for _, cols := range la.possibleProperties {
-		_, keys := getPermutation(cols, la.groupByCols)
-		if len(keys) != len(la.groupByCols) {
+
+	streamAggs := make([]PhysicalPlan, 0, len(la.possibleProperties)*(len(wholeTaskTypes)-1))
+	childProp := &requiredProp{
+		desc:        prop.desc,
+		expectedCnt: math.Max(prop.expectedCnt*la.inputCount/la.stats.count, prop.expectedCnt),
+	}
+
+	for _, possibleChildProperty := range la.possibleProperties {
+		sortColOffsets := getMaxSortPrefix(possibleChildProperty, la.groupByCols)
+		if len(sortColOffsets) != len(la.groupByCols) {
 			continue
 		}
-		for _, tp := range wholeTaskTypes {
-			// Second read in the double can't meet the stream aggregation's require prop.
-			if tp == copDoubleReadTaskType {
-				continue
-			}
-			childProp := &requiredProp{
-				taskTp:      tp,
-				cols:        keys,
-				desc:        prop.desc,
-				expectedCnt: prop.expectedCnt * la.inputCount / la.stats.count,
-			}
-			if childProp.expectedCnt < prop.expectedCnt {
-				childProp.expectedCnt = prop.expectedCnt
-			}
-			if !prop.isPrefix(childProp) {
-				continue
-			}
+
+		childProp.cols = possibleChildProperty[:len(sortColOffsets)]
+		if !prop.isPrefix(childProp) {
+			continue
+		}
+
+		for _, taskTp := range []taskType{copSingleReadTaskType, rootTaskType} {
```

Comment:
Add a comment such as:
" DoubleRead can't meet the stream aggregation's require prop. So we do not consider it here."